### ISSUE TITLE
Rector suggesting type-cast for backed enum

### ISSUE THREAD

**Issue by Knallcharge** [rectorphp/rector#8072]
I'm seeing a strange behavior/suggestion with the latest versions of rector (0.17.6, but it started one or two minor versions earlier). I have a couple of (symfony) commands that return an integer taken from a backed enum.
I tried to create a failing example but couldn't, rector always displays "// no change" - still, in my project, rector suggests adding a type-cast for the return value.

Code would be something like this:
```
enum ExitCode: int
{
    case Success = 0;
}

final class Command
{
    protected function test(): int {
        return ExitCode::Success->value;
    }
}
```

For each return I receive this:
![grafik](https://github.com/rectorphp/rector/assets/1715912/4aa99ade-d0d5-4e13-8bdb-e2c3b2078a64)

The strange part is that there are not "Applied rules" listed after each suggestion, rector just continues with the next file, so I can't even add the rule to my rector.php to ignore it.

Posting this as a question as I can't create reproducable code, I'd just like to know where this behavior may come from (especially the missing "Applied rules"-part).

**Comment by TomasVotruba** [comment]
Not sure what exact rule this is. It might help this will be related to some "cast" rule

**Comment by Knallcharge** [comment]
I've tried and disabled all rulesets, enabling them one by one and running rector again. The type-cast-recommendations occur as soon as I enable SymfonyLevelSetList::UP_TO_SYMFONY_63.
I tried reducing the setlist (down to SymfonyLevelSetList::UP_TO_SYMFONY_54), but the problem remained.

**Comment by TomasVotruba** [comment]
I see, we're getting to the rule :) 

Does this happen in UP_TO_SYMFONY_53 as well?
Which exact rule from UP_TO_SYMFONY_54 does it trigger?

**Comment by Knallcharge** [comment]
I have done some more tests in the meantime. On one run, rector showed "ReplaceArgumentDefaultValueRector" being the applied rule (which doesn't really make sense to me). Then I placed a return value into an entity-class, just to check if this had something to do with all effected classes being commands, but the type-cast is recommended here also. But: This time I got two other applied rules displayed: CommonNotEqualRector and AttributeKeyToClassConstFetchRector, which also don't seem to fit to the observed problem.
I will try to ignore the above rules and check again, whether the issue disappears.
Will try UP_TO_SYMFONY_53 and --debug afterwards.

**Comment by Knallcharge** [comment]
Tried using --debug, but didn't find anything. What am I looking for in the output? It just looks like this:

[file] src/Command/UpdateUserDataCommand.php
[rule] Rector\Renaming\Rector\Name\RenameClassRector
--- consumed 0 B, total 388 MB, took 0.00 s

[file] src/Command/UpdateUserDataCommand.php
[rule] Rector\Symfony\Symfony43\Rector\StmtsAwareInterface\TwigBundleFilesystemLoaderToTwigRector
--- consumed 0 B, total 388 MB, took 0.00 s

[file] src/Command/UpdateUserDataCommand.php
[rule] Rector\Symfony\Symfony42\Rector\New_\RootNodeTreeBuilderRector
--- consumed 0 B, total 388 MB, took 0.00 s

[file] src/Command/UpdateUserDataCommand.php
[rule] Rector\Renaming\Rector\Name\RenameClassRector
--- consumed 0 B, total 388 MB, took 0.00 s
(...)

This goes on for 2,7 million lines (literally, the textfile is 113MB), all entries are the same: "[file]...", "[rule]...", "-- consumed 0B (MB goes up slightly) took 0.00s" - followed by the same final result output as before (without showing "applied rules").

**Comment by Knallcharge** [comment]
Ok, trying the next thing: I ran rector again (without --dry-run this time) and dumped the debug-output to a file again, creating a >300MB file - still no difference in the listing of the files and rules and still no "applied rules" at the end where the type-casts are recommended.
I took one of the Commands that gets the type-cast-recommendations and copied all the rules from the dump that were applied to the file. Knowing that it came from a symfony-SetList, I removed all setlists and only included all rules starting with `Rector\Symfony` in my rector.php (using `$rectorConfig->rules([`) and removed the rules one by one, re-running rector each time, until the obsolete recommendations stopped.

Here's the rule that obviously causes this behavior: **Rector\Symfony\Symfony44\Rector\ClassMethod\ConsoleExecuteReturnIntRector::class**

Maybe the rule is too old (symfony 4.4) to understand enums in return-statements?
Anyways, ignoring that specific rule enables me to finally use rector again :)
